token_import.py

```diff
@@ -46,4 +26,27 @@ def read_used_list():
         print(f"{USED_FILE} not exist, create one!")
         f = open(USED_FILE, 'w+')
         f.close()
-        return list()
\ No newline at end of file
+        return list()
+
+def _import_qrcode_from_bigquery(token_dict: dict, table_name: str) -> dict[str, str]:
+    current_year = datetime.now().year
+    query = f"""
+        SELECT
+          ticket_type,
+          qrcode
+        FROM
+          `pycontw-225217.{table_name}`
+        WHERE
+          qrcode IS NOT NULL
+          AND payment_status = 'paid'
+          AND EXTRACT(YEAR
+          FROM
+            CAST(paid_date AS DATE)) = {current_year};
+    """
+    query_job = BIQUERY_CLIENT.query(query)
+    for row in query_job:
+        ticket_name = row["ticket_type"].lower().replace("\"", "")
+        qrcode = row["qrcode"].replace("\"", "")
+        assert len(qrcode) == 32, f"qrcode length should be 32, but got {len(qrcode)}"
+        token_dict[qrcode] = ticket_name
+    return token_dict
```

Comment:
1. read data from BigQuery
2. check if they're valid qrcodes
3. update `token_dict` (schema of `token_dict` remains the same, just changed the way we get those qrcode data)